test/nonStrict.test.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  expectAcceptValuesImpure,
+  expectRejectValues,
+  objectAttributes,
+  st,
+} from './helpers'
+
+describe('nonStrict', () => {
+  it('accepts records', () => {
+    const runtype = st.nonStrict(
+      st.record({
+        a: st.integer(),
+        b: st.string(),
+      }),
+    )
+
+    // a nonStrict runtype is always impure
+    expectAcceptValuesImpure(
+      runtype,
+      [
+        [
+          { a: 0, b: 'foo', c: true }, // ignores additional keys
+          { a: 0, b: 'foo' },
+        ],
+      ],
+      true,
+    )
+  })
+
+  it('accepts nonStrict records', () => {
+    const runtype = st.nonStrict(
+      st.nonStrict(
+        st.record({
+          a: st.integer(),
+          b: st.string(),
+        }),
+      ),
+    )
+
+    expectAcceptValuesImpure(
+      runtype,
+      [
+        [
+          { a: 0, b: 'foo', c: true },
+          { a: 0, b: 'foo' },
+        ],
+      ],
+      true,
+    )
+  })
+
+  it('accepts records with optional values', () => {
+    const runtype = st.nonStrict(
+      st.record({
+        a: st.integer(),
+        b: st.optional(st.string()),
+      }),
+    )
+
+    expectAcceptValuesImpure(
+      runtype,
+      [
+        [
+          { a: 0, b: 'foo', c: true },
+          { a: 0, b: 'foo' },
+        ],
+        [
+          { a: 0, b: undefined, c: true }, // ignores additional keys
+          { a: 0, b: undefined },
+        ],
+      ],
+      true,
+    )
+  })
+
+  it('accepts nested records and does not apply recursively', () => {
+    const runtype = st.nonStrict(
+      st.record({
+        a: st.record({
+          b: st.record({
+            c: st.string(),
+          }),
+        }),
+      }),
+    )
+
+    expectAcceptValuesImpure(
+      runtype,
+      [[{ a: { b: { c: 'foo' } }, d: 0 }, { a: { b: { c: 'foo' } } }]],
+      true,
+    )
+    expectRejectValues(runtype, [
+      { a: { b: { c: 'foo' }, d: 0 } },
+      { a: { b: { c: 'foo', d: 0 } } },
+    ])
+  })
+
+  it('rejects runtypes that are not records', () => {
```

Comment:
Would like to discuss whether all that is in this test is correct / how you expected `nonStrict` to work (and not work).

Reply:
we also don't currently have a test helper that does what I'm doing in this test: that is, take in multiple runtypes and check their usage in another. the `expectXxxValues` helpers test multiple values against one runtype, where here I'm testing multiple runtypes. not sure if it's worth adding a helper (would need to check if there were other use cases)